textdescriptives/extractor.py

```diff
@@ -0,0 +1,100 @@
+"""Extract metrics as Pandas DataFrame"""
+from spacy.tokens import Doc
+
+from typing import Union
+import types
+
+import pandas as pd
+
+
+class Extractor:
+    def __init__(
+        self,
+        doc: Doc,
+        metrics: Union[list[str], str] = "all",
+        include_text: bool = True,
+    ):
+        """Utility class to extract specified metrics to a Pandas DataFrame
+
+        Args:
+            doc (Doc): a spaCy doc
+            metrics (Union[list[str], str], optional): Which metrics to extract.
+                One or more of ["descriptive_stats", "readability", "dependency_distance", "all"].
+                Defaults to "all".
+            include_text (bool, optional): Whether to add a column containing the text. Defaults to True.
+        """
+        if not isinstance(doc, (Doc)):
+            raise TypeError(f"doc should be a spaCy Doc object, not {type(doc)}.")
+
+        valid_metrics = set(
+            ["descriptive_stats", "readability", "dependency_distance", "all"]
+        )
+        if not isinstance(metrics, (str, list)):
+            raise TypeError(
+                f"'metrics' should be string or list of strings, not {type(metrics)}"
+            )
+        if isinstance(metrics, str):
+            metrics = [metrics]
```

Comment:
to not check str type twice

```suggestion
        if isinstance(metrics, str):
            metrics = [metrics]
        if not isinstance(metrics, list):
            raise TypeError(
                f"'metrics' should be string or list of strings, not {type(metrics)}"
            )
```